test/integ/test_key_pair_authentication.py

```diff
@@ -14,6 +17,53 @@
 import snowflake.connector
 
 
+@pytest.mark.skipolddriver
+@pytest.mark.parametrize(
+    "input_account,expected_account",
+    [
+        ("s3testaccount.global", "S3TESTACCOUNT.GLOBAL"),
+        ("acct-with-dashes", "ACCT_WITH_DASHES"),
+        ("testaccount.extra", "TESTACCOUNT"),
+        ("testaccount-user.global", "TESTACCOUNT"),
+        ("normalaccount", "NORMALACCOUNT"),
+    ],
+)
+def test_get_token_from_private_key(input_account, expected_account):
+    test_user = "python_test_keypair_user_" + str(uuid.uuid4()).replace("-", "_")
+    current_dir = path.dirname(path.realpath(__file__))
+    private_key_file_path = path.join(
+        current_dir, "..", "data", "rsa_keys", "rsa_key_encrypted.p8"
+    )
+    private_key_password = "test"
+    public_key_fingerprint = snowflake.connector.auth.get_public_key_fingerprint(
+        private_key_file_path, private_key_password
+    )
+    # generate the jwt token
+    jwt_token = snowflake.connector.auth.get_token_from_private_key(
+        test_user, input_account, private_key_file_path, private_key_password
+    )
+    # decode the token to get its fields (iss, sub, issue time, expiration time)
+    decoded_token = jwt.decode(jwt_token, options={"verify_signature": False})
+    # Assert "sub" field matches {corrected account}.{user}
+    assert expected_account + "." + test_user.upper() == decoded_token.get("sub")
+    # Assert "iss" field matches {corrected account}.{user}.{public key fingerprint}
+    assert (
+        expected_account
+        + "."
+        + test_user.upper()
+        + "."
+        + public_key_fingerprint.upper()
```

Comment:
This is more of an FYI, don't do a new commit for this only.
There are these cool new features called [f-string](https://www.python.org/dev/peps/pep-0498/)
```suggestion
        f"{expected_account}.{test_user.upper()}.{public_key_fingerprint.upper()}"
```